### DISCUSSION TITLE
How to change button color?

### DISCUSSION THREAD

**Discussion by dr-sandipan-chowdhury** [zauberzeug/nicegui#2297] [Q&A]
### Question

I am new to NICEGUI. Can anyone please help me with how to change button color? 
`ui.button(text=s).props("flat").style("color:black;")`

**Comment by dr-sandipan-chowdhury** [comment]
Now I get it. css and tailwind seems to have no effect. props("color=black") is working.

**Comment by stepheweffie** [comment]
Changing 'color' in CSS .style will change the font color. You can access the Quasar based color palette in NiceGUI with '.classes('bg-primary')' and "primary" is the component color name which can be changed for one of these, e.g. "secondary", "info", "dark" - 
[https://quasar.dev/style/color-palette#color-list](url)
look at theme builder too 
[https://quasar.dev/style/theme-builder](url)

**Comment by falkoschindler** [comment]
You're right, CSS and Tailwind are kind of overruled by Quasar's button styling. But using Quasar's props is one way to change the color.

There's also a `color` parameter:
```py
ui.button('Click me', color='red')
```
<img width="110" alt="Screenshot 2024-01-06 at 12 01 05" src="https://github.com/zauberzeug/nicegui/assets/5767091/50ff97af-57ad-4be2-9b6e-0b524c10be1e">

**Comment by amks1** [comment]
Is there a way to change the color on hover? I didn't find anything in Quasar's button documentation.

**Reply by falkoschindler** [reply]
See https://github.com/zauberzeug/nicegui/discussions/2943

**Comment by ipsi-apant** [comment]
I tried this with tailwind colors. It is not applying colors.

```python
ui.button('Button').classes('bg-slate-100 text-slate-950') # Not working. This is setting css `bg-color` prop

ui.button('Button').classes('bg-red') # Working! This is setting css `background` prop

ui.run(dark=None)
```

**Reply by falkoschindler** [reply]
Quasar tends to overrule Tailwind classes. You can add a `!` prefix to mark the CSS property as [important](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_!important_exception):
```py
ui.button('Button').classes('!bg-slate-100 !text-slate-950')
```

**Reply by ipsi-apant** [reply]
Love it! This is working! Appreciate it @falkoschindler

**Reply by Twangist** [reply]
Thanks for this. When changing button color dynamically, I also found it necessary to mark as !important the bg color class to be `remove`d as well as for the new color. Thus, 

```py
btn = ui.button('Yo').classes('!bg-red-400')
# ...
# Change color:
btn.classes('!bg-yellow-400', remove='!bg-red-400')
```
